tests/test_spectrum_assignment.py

```diff
@@ -0,0 +1,378 @@
+    print('initial spectrum')
+    print(nvalue_to_frequency(oms_list[10].spectrum_bitmap.freq_index_min) * 1e-12,
+          nvalue_to_frequency(oms_list[10].spectrum_bitmap.freq_index_max) * 1e-12)
+    # checks initial values consistancy
+    ind_max = len(oms_list[10].spectrum_bitmap.bitmap) - 1
+    print('with guardband', oms_list[10].spectrum_bitmap.getn(0),
+          oms_list[10].spectrum_bitmap.getn(ind_max))
+    print('without guardband', oms_list[10].spectrum_bitmap.freq_index_min,
+          oms_list[10].spectrum_bitmap.freq_index_max)
+    test1 = oms1.assign_spectrum(nval1, mval)
+    print(oms1.spectrum_bitmap.bitmap)
+    # if requested slots exceed grid spectrum should not be assigned and assignment
+    # should return False
+    if ((nval1 - mval) < oms1.spectrum_bitmap.getn(0) or
+            (nval1 + mval-1) > oms1.spectrum_bitmap.getn(ind_max)):
+        print('assignment on part of bitmap is not allowed')
+        assert not test1
+        for elem in oms1.spectrum_bitmap.bitmap:
+            assert elem == 1
+    else:
+        oms2.assign_spectrum(nval2, mval)
+        print(oms2.spectrum_bitmap.bitmap)
+        test2 = bitmap_sum(oms1.spectrum_bitmap.bitmap, oms2.spectrum_bitmap.bitmap)
+        print(test2)
+        range1 = range(oms1.spectrum_bitmap.geti(nval1) - mval,
+                       oms1.spectrum_bitmap.geti(nval1) + mval -1)
+        range2 = range(oms2.spectrum_bitmap.geti(nval2) - mval,
+                       oms2.spectrum_bitmap.geti(nval2) + mval -1)
+        for elem in range1:
+            print(f'value should be zero at index {elem}')
+            assert test2[elem] == 0
+
+        for elem in range2:
+            print(f'value should be zero at index {elem}')
+            assert test2[elem] == 0
+
+def test_values(setup):
+    """ checks that oms.assign_spectrum(13,7) is (193137500000000.0, 193225000000000.0)
+        reference to Recommendation G.694.1 (02/12), Figure I.3
+        https://www.itu.int/rec/T-REC-G.694.1-201202-I/en
+    """
+    network, oms_list = setup
+
+    oms_list[5].assign_spectrum(13, 7)
+    fstart, fstop = m_to_freq(13, 7)
+    print('expected: 193137500000000.0, 193225000000000.0')
+    print(f'obtained: {fstart}, {fstop}')
+    assert fstart == 193.1375e12 and fstop == 193.225 * 1e12
+    nstart = frequency_to_n(fstart)
+    nstop = frequency_to_n(fstop)
+    # nval, mval = slots_to_m(7, 20)
+    nval, mval = slots_to_m(nstart, nstop)
+    print('expected n, m: 13, 7')
+    print(f'obtained: {nval}, {mval}')
+    assert nval == 13 or mval == 7
+
+@pytest.mark.parametrize('nval', [0, None, 0.5])
+@pytest.mark.parametrize('mval', [1, 0, None, 4.5])
+def test_exception(nval, mval, setup):
+    """ test n or m not applicable values
+    """
+    network, oms_list = setup
+    try:
+        oms_list[5].assign_spectrum(nval, mval)
+        print(f'n, m values should raise an error {nval}, {mval}')
+        test = False
+    except SpectrumError:
+        test = True
```

Comment:
Pytest has a way of saying that [this code should always throw](https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions), that should be used here.

This test is, however, a bit more complicated because it is somehow expected to *not* fail when `m + n = 1`. That's non-obvious to me, and I think it would be better to split this into two separate tests, one for correct values which should not throw, and one for expected failures.

Reply:
Thanks! good to know! also enabled me to add a test on negative values for m (not allowed) :)